packages/analytics-js/src/services/StoreManager/StoreManager.ts

```diff
@@ -70,49 +79,79 @@ class StoreManager implements IStoreManager {
    * Create store to persist data used by the SDK like session, used details etc
    */
   initClientDataStore() {
-    const storageType = state.storage.type.value || COOKIE_STORAGE;
-    let finalStorageType = storageType;
-
-    switch (storageType) {
-      case LOCAL_STORAGE:
-        if (!getStorageEngine(LOCAL_STORAGE)?.isEnabled) {
-          finalStorageType = MEMORY_STORAGE;
-        }
-        break;
-      case MEMORY_STORAGE:
-        finalStorageType = MEMORY_STORAGE;
-        break;
-      case NO_STORAGE:
-        finalStorageType = NO_STORAGE;
-        break;
-      case COOKIE_STORAGE:
-      default:
-        // First try setting the storage to cookie else to local storage
-        if (getStorageEngine(COOKIE_STORAGE)?.isEnabled) {
-          finalStorageType = COOKIE_STORAGE;
-        } else if (getStorageEngine(LOCAL_STORAGE)?.isEnabled) {
-          finalStorageType = LOCAL_STORAGE;
-        } else {
-          finalStorageType = MEMORY_STORAGE;
-        }
-        break;
+    const globalStorageType = state.storage.type.value;
+    const storageTypesRequiringInitialization = [MEMORY_STORAGE];
+    if (getStorageEngine(LOCAL_STORAGE)?.isEnabled) {
+      storageTypesRequiringInitialization.push(LOCAL_STORAGE);
     }
-
-    if (finalStorageType !== storageType) {
-      this.logger?.warn(STORAGE_UNAVAILABLE_WARNING(STORE_MANAGER, storageType, finalStorageType));
+    if (getStorageEngine(COOKIE_STORAGE)?.isEnabled) {
+      storageTypesRequiringInitialization.push(COOKIE_STORAGE);
     }
 
+    let trulyAnonymousTracking = true;
+    const entries = state.loadOptions.value.storage?.entries;
+    Object.keys(state.storage.entries.value).forEach(entry => {
```

Comment:
You can directly iterate the storage entries enum.

Reply:
Comment not clear to me.